src/CoreWCF.Primitives/src/CoreWCF/IdentityModel/Selectors/SecurityTokenAuthenticator.cs

```diff
@@ -21,6 +23,12 @@ public bool CanValidateToken(SecurityToken token)
         }
 
         public ReadOnlyCollection<IAuthorizationPolicy> ValidateToken(SecurityToken token)
+        {
+            // We don't call this internally, so we won't have problems with sync over async causing performance issues
+            return ValidateTokenAsync(token).AsTask().GetAwaiter().GetResult();
+        }
```

Comment:
Try to avoid allocation when it completes synchronouslys.
```c#
var validationResult = ValidateTokenAsync(token);
if (validationResult.IsCompleted)
{
    return validationResult.Result;
}
else
{
    return validationResult.AsTask().GetAwaiter().GetResult();
}

Reply:
done